### ISSUE TITLE
Results giving weird look after using your shader.

### ISSUE THREAD

**Issue by bodduavinash** [ColinLeung-NiloCat/UnityURPUnlitScreenSpaceDecalShader#30]
Results giving weird look after using your shader.

I have followed the instructions given in readme file, still giving weird looks.

Please see the below screenshot, for better understanding.

![image](https://user-images.githubusercontent.com/3116586/111898421-fc007200-8a4b-11eb-845c-7d6fb4b64e3c.png)

Kindly help me solve this issue.

Appreciate if anyone faced and help me for this issue.

**Comment by ColinLeung-NiloCat** [comment]
turn on _ProjectionAngleDiscardEnable

**Comment by bodduavinash** [comment]
Thank you for the response,

I've turned on _ProjectionAngleDiscardEnable, below is the result after setting threshold value to 0.02, but the bottom is not projecting right.

![image](https://user-images.githubusercontent.com/3116586/111899450-b09d9200-8a52-11eb-943d-7fbffe94fc16.png)


![image](https://user-images.githubusercontent.com/3116586/111899447-a9768400-8a52-11eb-900e-964f828b84ee.png)

My expectation result should be something below.

![image](https://user-images.githubusercontent.com/3116586/111899498-f22e3d00-8a52-11eb-8441-14b4ce6d6620.png)


Is there any other settings do I need to apply? Kindly help me solve this issue.

**Comment by bodduavinash** [comment]
@ColinLeung-NiloCat 
> Thank you for the response,
> 
> I've turned on _ProjectionAngleDiscardEnable, below is the result after setting threshold value to 0.02, but the bottom is not projecting right.
> 
> ![image](https://user-images.githubusercontent.com/3116586/111899450-b09d9200-8a52-11eb-943d-7fbffe94fc16.png)
> 
> ![image](https://user-images.githubusercontent.com/3116586/111899447-a9768400-8a52-11eb-900e-964f828b84ee.png)
> 
> My expectation result should be something below.
> 
> ![image](https://user-images.githubusercontent.com/3116586/111899498-f22e3d00-8a52-11eb-8441-14b4ce6d6620.png)
> 
> Is there any other settings do I need to apply? Kindly help me solve this issue.

**Comment by ColinLeung-NiloCat** [comment]
rotate the decal gameobject a bit

**Comment by bodduavinash** [comment]
@ColinLeung-NiloCat 
Yes, I have tried X rotation about 20 units, then it started showing better results.

![image](https://user-images.githubusercontent.com/3116586/111904603-c2d8f980-8a6d-11eb-9262-926ce7443929.png)

I am implementing FPS shooter game. 

If I shoot straight wall or ground alone it works well, but if I shoot corner of both straight wall or ground, then how can I detect that bullet is hitting at corner? in other words, when should I rotate if the bullet is hitting the corner? Is there any workaround for this? Kindly help me to solve this issue.

Thanks,
Avinash.
